diesel/src/sqlite/expression/collate.rs

```diff
@@ -0,0 +1,101 @@
+use backend::Backend;
+use expression::*;
+use query_builder::*;
+use result::QueryResult;
+
+pub trait Collation {
+    fn get_collation<'a>(&'a self) -> &'a str;
```

Comment:
I think this should be a associated constant instead of a member function.
```rust
const COLLATION: &'static str;
```

Reply:
I thought about this too, but then I found out that other DB engines also have `COLLATE`, and it's used a little differently.
https://www.postgresql.org/docs/9.6/static/collation.html
https://dev.mysql.com/doc/refman/5.7/en/charset-collate.html

If this trait will be reused for them, then it's probably better to allow non-static strings here.

Reply:
This trait would need to be generic over the backend for it to be used by more than just SQLite. If we think we will want to re-use this for other backends, we need to do that in this PR. We're a 1.0 library, so we can't change this later.